doc/rst/technotes/overloadSets.rst

```diff
@@ -0,0 +1,58 @@
+.. _readme-overload-sets:
+
+======================
+Checking Overload Sets
+======================
+
+The Chapel compiler provides an initial implementation of the
+Overload Sets feature. It is discussed in `CHIP 20
+<https://github.com/chapel-lang/chapel/blob/master/doc/rst/developer/chips/20.rst>`_
+as a solution to "hijacking" problems where program behavior can change
+unexpectedly when imported modules change.
+
+An "overload set" is a group of functions with the same name declared
+in the same module. Overload sets checking is performed at each
+function call. It verifies that all candidate functions identified
+while resolving this call are defined in the same module. See the
+"Function Resolution" section of the language spec for
+resolution-related terminology.
+
+Additionally, if the call resolves to "ref-pair" overloads,
```

Comment:
I don't think ref pair is a term in the spec. I thought we were just calling it "return intent overloading", so here you could say "if the call resolves to return intent overloads"